uefi-test-runner/src/proto/shell.rs

```diff
@@ -0,0 +1,40 @@
+use uefi::{prelude::*, proto::{shell::Shell, media::fs::SimpleFileSystem, device_path::{build::DevicePathBuilder, media::FilePath}}, table::boot::{SearchType, LoadImageSource}};
```

Comment:
It looks like this needs to run through `rustfmt --all`. Also, `cargo xtask clippy` will give a few warnings.

Reply:
Yup, the last commit is still very messy, it was mostly pushed to get feedback on the FileList struct and its soundness. Thanks for looking over it though :)